pkg/provider/vault/auth.go

```diff
@@ -160,6 +161,16 @@ func checkToken(ctx context.Context, token util.Token) (bool, error) {
 	if tokenType == "batch" {
 		return false, nil
 	}
+	ttl, ok := resp.Data["ttl"]
+	if !ok {
+		return false, fmt.Errorf("no TTL found in response")
+	}
+	ttlInt, err := ttl.(json.Number).Int64()
+	if err != nil {
+		return false, fmt.Errorf("invalid token TTL: %v: %w", ttl, err)
+	} else if ttlInt < 60 {
+		return false, fmt.Errorf("token expiring in %vs; treating as expired", ttlInt)
+	}
```

Comment:
```suggestion
	ttlInt, err := ttl.(json.Number).Int64()
	if err != nil {
		return false, fmt.Errorf("invalid token TTL: %v: %w", ttl, err)
	} 
	creationTtl, ok := resp.Data["creation_ttl"]
	if !ok {
	  		return false, fmt.Errorf("no TTL found in response")
	      }
	creationTtlInt, err := creationTtl.(json.Number).Int64()
	      
	if ttlInt < 60 && creationTtlInt > 0 {
		return false, fmt.Errorf("token expiring in %vs; treating as expired", ttlInt)
	}
```

Reply:
might be good to validate the logic, coding on github UI is terrible 😄

Reply:
actually realized I found a neater way. `expire_time` will be `null` whenever we don't need to do the check. So we can simply add a `if expireTime == nil { return true }` block